honeycombio/resource_query_annotation.go

```diff
@@ -28,12 +29,14 @@ func newQueryAnnotation() *schema.Resource {
 				ForceNew: true,
 			},
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 120),
 			},
 			"description": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 280),
```

Comment:
Just to be certain: the `Optional` here overrides the validation such that an empty description is okay?

Reply:
You got it! and, if it is present it must be between 1 and 280 chars